### ISSUE TITLE
Adding a domain before a build/release fails

### ISSUE THREAD

**Issue by helgi** [deis/workflow#317]
RCs are not created until build/release time

Create App
```
deis create --no-remote
Creating Application... done, created yearly-earthman
remote available at ssh://git@deis.172.17.8.100.xip.io:2222/yearly-earthman.git
```

Add Domain
```
deis domains:add foo.com -a yearly-earthman
Adding foo.com to yearly-earthman... Error:
503 Service Unavailable
detail: failed to get Service "yearly-earthman" in Namespace "yearly-earthman": 404 Not Found
{'apiVersion': 'v1', 'metadata': {}, 'code': 404, 'status': 'Failure', 'reason': 'NotFound', 'kind': 'Status', 'details': {'kind': 'services', 'name': 'yearly-earthman'}, 'message': 'services "yearly-earthman" not found'}
```

**Comment by carmstrong** [comment]
Is this also the case with `config:set`? I know this was an issue with our old model of representing apps - some things can't be configured until there's a release.

**Comment by helgi** [comment]
It is the case with `config:set` as well, both will try to interact with a RC before it exists, mostly due to how we lazy create the RC (during a deploy), but we can't do it ahead of time since RC = release.

**Comment by helgi** [comment]
The domain specific issue was resolved with #473
